cpp/src/io/orc/writer_impl.cu

```diff
@@ -2202,9 +2213,6 @@ stripe_dictionaries build_dictionaries(orc_table_view& orc_table,
   gpu::collect_map_entries(stripe_dicts, stream);
   gpu::get_dictionary_indices(stripe_dicts, orc_table.d_columns, stream);
 
-  // deallocate hash map storage, unused after this point
-  hash_maps_storage.clear();
```

Comment:
Can we still deallocate `map_storage` early?

Reply:
I tried to but couldn't find an `cuco::aow_storage` API to do so

Reply:
If not, I can also just scope `map_storage` such that it is destroyed here.

Reply:
is that doable? that would be the cleanest option IMO

Reply:
Use `unique_ptr`:
```cpp
auto map_storage = std::make_unique<storage_type>( ... );
...
map_storage.reset(); // the same as hash_maps_storage.clear();
```

Reply:
All done!

Reply:
Let me change to `unique_ptr`

Reply:
All done!